encoder.go

```diff
@@ -0,0 +1,143 @@
+package zapvml
+
+import (
+	"regexp"
+	"time"
+
+	"go.uber.org/zap/buffer"
+	"go.uber.org/zap/zapcore"
+)
+
+func newEncoder(cfg zapcore.EncoderConfig) (zapcore.Encoder, error) {
+	return &encoder{zapcore.NewJSONEncoder(cfg)}, nil
+}
+
+// Wraps zapcore.Encoder to customize stack traces to be picked up by Stackdriver error reporting.
+// The following issue might make this unnecessary:
+// https://github.com/uber-go/zap/issues/514
+type encoder struct {
+	jsonEncoder zapcore.Encoder
+}
+
+// multiline pattern to match the function name line
+var functionNamePattern = regexp.MustCompile(`(?m)^(\S+)$`)
+
+func (s *encoder) EncodeEntry(ent zapcore.Entry, fields []zapcore.Field) (*buffer.Buffer, error) {
+	if ent.Level == zapcore.ErrorLevel && ent.Stack != "" {
```

Comment:
Could you explain this? I think we should not pretend something is a panic that isn't. Stackdriver should probably only pick up errors that are actually serious.

Reply:
@lyschoening Nothing is pretending to be a panic. This is a way of getting error stack trace into the correct format so GCP error reporting can pick it up.

Reply:
Alright. Got confused by the comments :)

Reply:
@lyschoening  I tried to make it prettier but it seems that GCP error reporting is very sensitive when it comes to stack trace formatting 😄

Reply:
It would be nice if we could use the method they expect. They referenced it somewhere. But I guess it would be tricky to reuse.

Reply:
@lyschoening they just mention this https://cloud.google.com/error-reporting/reference/rest/v1beta1/projects.events/report?hl=en_GB#reportederrorevent